src/lib/tuning/abstract_selector.hpp

```diff
@@ -0,0 +1,34 @@
+#pragma once
+
+#include <memory>
+#include <vector>
+
+#include "tuning/tuning_choice.hpp"
+#include "tuning/tuning_operation.hpp"
+
+namespace opossum {
+
+/**
+ * An AbstractSelector transforms an unordered list of TuningChoices into
+ * a concrete sequence of TuningOperations.
+ *
+ * It considers a cost budget that the entire operation sequence as well as any
+ * continuous subsequence from the beginning must not exceed.
+ * The operation sequence is prioritized by expected performance impact on the system,
+ * so that the most beneficial operations come before less useful operations.
+ *
+ * For more information on the underlying problem, see e.g. Wikipedia:
```

Comment:
Not all of these problems are reducible to the KP (the KP does not consider interaction of elements). Please add a note regarding this simplification.

Reply:
done